### ISSUE TITLE
Limit this package to ST2 and ST3?

### ISSUE THREAD

**Issue by deathaxe** [braver/TypeScriptSyntax#8]
As ST4 provides a very good Typescript syntax, it may make sense to limit this package to ST2 and 3 on Package Control.

**Comment by braver** [comment]
Sure, good idea!

**Comment by patrickfatrick** [comment]
Why though? Any color schemes built with this syntax in mind will look different with Sublime's default Typescript syntax.

**Comment by deathaxe** [comment]
ST's syntax definitions are far superior and the intellisense features often cause issues and user complaints. The general purpose recommendation is to use LSP and LSP-typescript packages.

**Comment by patrickfatrick** [comment]
This is just a syntax, not the language server features of the full Typescript plugin. I agree that one should use LSP over the Typescript plugin but the point of this package is to provide _just_ the syntax that's provided by the Typescript plugin (so you don't have to use the Typescript plugin), and if it works fine in ST4 I see no reason the author should arbitrarily limit it to <= ST3.

edit: Unless either this package or the Typescript plugin have been deprecated, of course...

**Comment by deathaxe** [comment]
Sorry, haven't read the title carefully enough. My statement about the syntax is still valid, but the code intellisense features targeted https://github.com/wbond/package_control_channel/pull/8479

**Comment by braver** [comment]
A syntax in ST is never “just a syntax”. Also, consistency with other syntaxes is important. This isn’t just for the colors. Therefore ST4 users really should never use this package.

**Comment by patrickfatrick** [comment]
> My statement about the syntax is still valid

As an end user that seems debatable to me? They both work well enough as far as I've seen. In some ways I prefer this syntax. Maybe I'm missing something?

> Therefore ST4 users really should never use this package.

And yet I still don't really understand what the rationale is. Consistency with the core syntaxes is not a great reason because that's why custom syntaxes exist in the first place. So is the core syntax is objectively better than this one for some reason? Is the MS syntax not supported in ST4 or missing features at this point? I ask because I just want to understand, I was using this syntax just fine in ST4 (with a personal color scheme I maintain) so from my POV this just seems like an arbitrary decision to remove the choice from the end user.

But as you are the author I will have to defer to your judgement and will probably just switch to JS Custom for Typescript usage.

**Comment by braver** [comment]
> Consistency with the core syntaxes is not a great reason because that's why custom syntaxes exist in the first place

Not really actually. Most custom syntaxes were created because there wasn't an excellent default syntax out there. And then they stick around. Just like this one. Usually for a syntax it's not a question of taste. 

A syntax does its best to describe to the editor how the code is structured. The editor, and various plugins, can then use that information to create functionality like goto-definition, list symbols, find usages, autocomplete suggestions, all without necessarily leaning on something like a language service. So consistency is key here: the more consistent, the better the features. Also the more the colors match with what you're seeing in other languages, which is also important. A syntax should never change the way they represent a piece of code just so it looks better, that's for the color scheme to decide. And again, the more well defined a syntax is and the more it is consistent with others, the better the color schemes. A syntax in ST4 is not about the colors, that's just one of the many side effects.
 
I have never seen community syntaxes as well developed, maintained and scrutinized as the default syntaxes that ship with Sublime Text 4. This includes what I've seen in other editor eco systems. Sure, interest in some area might spike temporarily for a given editor, but the effort in ST4 syntaxes is sustained at a very high level. 
Specifically for TypeScript, this somewhat half-assed job that Microsoft did ages ago might very well work for you, but as a syntax within the ST4 ecosystem it is very flawed and can't hold a candle to what is currently shipping with ST4 by default.

> So is the core syntax is objectively better than this one for some reason? 

In short yes, because it adheres closely to the guidelines for describing a language and does a very good and detailed job.

**Comment by braver** [comment]
oh, by the way, this is already done:
https://github.com/wbond/package_control_channel/blob/master/repository/t.json#L3602
